src/lib/entry/ServiceTestFixture.ts

```diff
@@ -0,0 +1,111 @@
+import {Store} from 'redux'
+import {createMemoryHistory, History, Location} from 'history'
+import {filter} from 'lodash'
+import {PluginConfig, PluginConstructor} from '../core/PluginConfig'
+import {ApplicationContext} from '../core/ApplicationContext'
+import {loadService} from '../core/load'
+import {createBrowserPlugin} from '../plugins/BrowserPlugin/BrowserPlugin'
+import {mountSpy} from '../plugins/ControllerMountSpyPlugin/ControllerMountSpyPlugin'
+import {gatherServices, Service, ServiceConstructor} from '../core/Service'
+
+export interface ServiceTestFixtureProps {
+  plugins?: PluginConstructor[]
+  location?: string
+  hostname?: string
+  host?: string
+  protocol?: string
+  port?: number
+  service: ServiceConstructor
+}
+
+export class ServiceTestFixture<ServiceType extends Service> {
+  readonly history: History = createMemoryHistory()
+  private appContext: ApplicationContext
+  service: ServiceType
+  readonly serviceConstructor: ServiceConstructor
+
+  static async createMountedService<ServiceType extends Service>(props: ServiceTestFixtureProps): Promise<ServiceTestFixture<ServiceType>> {
+    const instance = new ServiceTestFixture<ServiceType>(props)
+    await instance.load()
+    return instance
+  }
+
+  private constructor({
+                        service: serviceConstructor,
+                        plugins = [],
+                        location = '/',
+                        hostname = 'example.com', host = 'example.com', protocol = 'http', port = 80
+                      }: ServiceTestFixtureProps) {
+    this.history.push(location)
+    this.appContext = new ApplicationContext([
```

Comment:
Is this just a starting context for the tests that use this fixture? Could this be extracted to a method that is named to explain what it is and what, if any, bits of it are important? getDefaultTestContext? getContextWith[X]Plugin?

Reply:
Yup. Added a better name